core/src/main/java/jeeves/transaction/TransactionManager.java

```diff
@@ -105,65 +168,255 @@
             } else if (exception[0] instanceof Error) {
                 throw (Error) exception[0];
             } else {
-                throw new RuntimeException(exception[0]);
+                throw new RuntimeException("Run in transaction '"+action+"': "+exception[0].getLocalizedMessage(), exception[0]);
             }
         }
         return result;
     }
 
-    protected static void doCommit(ApplicationContext context, PlatformTransactionManager transactionManager, TransactionStatus transaction) {
+    /**
+     * Safely perform {@link PlatformTransactionManager#commit(TransactionStatus)}.
+     *
+     * This method is responsible for safely logging any notification failures as warnings, rather than failing outright.
+     *
+     * @param context context used to obtain listeners to notify
+     * @param transactionManager manager responsible for execution
+     * @param status transaction status
+     */
+    protected static void doCommit(ApplicationContext context, PlatformTransactionManager transactionManager, TransactionStatus status) {
+        try {
+            fireBeforeCommit(context, transactionManager, status);
+        }
+        catch( Throwable t ){
+            Log.warning(Log.JEEVES, "Commit transaction - before:", t);
```

Comment:
This is a general pattern, making sure to catch and log any exceptions that occur during firing of notifications.

This should not be a behaviour change, if any listener was depending on throwing an exception to stop a commit ... that would be awkward API design and I would expect javadocs to explain the api contract.